marklogic-data-hub/src/test/java/com/marklogic/hub/impl/ClearUserModulesTest.java

```diff
@@ -0,0 +1,83 @@
+package com.marklogic.hub.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.client.admin.QueryOptionsManager;
+import com.marklogic.client.admin.ResourceExtensionsManager;
+import com.marklogic.client.admin.ServerConfigurationManager;
+import com.marklogic.client.admin.TransformExtensionsManager;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.QueryOptionsListHandle;
+import com.marklogic.hub.AbstractHubCoreTest;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ClearUserModulesTest extends AbstractHubCoreTest {
+    ClearUserModulesTest() {
+    }
+
+    @BeforeEach
+    void setUp() throws IOException {
+        Assumptions.assumeTrue(this.isVersionCompatibleWith520Roles());
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
It looks like this is setting up a project with a user module of each type. Just make a new test-project instead - e.g. test-project/all-module-types. And then add a file for each module type. That avoids all of this custom code, you can just use installProjectInFolder.